fetcher/fetcher_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fetcher
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/state"
+)
+
+// Setup parent immutable state
+var _ state.Immutable = &testDB{}
+
+type testDB struct {
+	storage map[string][]byte
+}
+
+func newTestDB() *testDB {
+	return &testDB{
+		storage: make(map[string][]byte),
+	}
+}
+
+func newTestDBWithValue() *testDB {
+	db := testDB{storage: make(map[string][]byte)}
+	for i := 0; i < 100; i++ {
+		if i%2 == 0 {
+			db.storage[strconv.Itoa(i)] = []byte("value")
+		}
+	}
+	return &db
+}
+
+func (db *testDB) GetValue(_ context.Context, key []byte) (value []byte, err error) {
+	val, ok := db.storage[string(key)]
+	if !ok {
+		return nil, database.ErrNotFound
+	}
+	return val, nil
+}
+
+func TestFetchSameKeysConcurrentBasic(t *testing.T) {
```

Comment:
Would be a good idea to incorporate the executor and pass `f.Wait()` into `e.Run(..)` for better testing

Reply:
Think this is indirectly blocked by #715 since the executor still doesn't know how to handle r/w keys in `main`

Reply:
For now, the unit tests in `fetcher_test.go` will closely mimic its usage in executor using go routines